src/winutils.cc

```diff
@@ -0,0 +1,126 @@
+
+
+#include "winutils.h"
+#include "TlHelp32.h"
+#include <cassert>
+
+
+DWORD getppid() {
+    HANDLE hSnapshot = INVALID_HANDLE_VALUE;
+    PROCESSENTRY32 pe32;
+    DWORD ppid = 0, pid = GetCurrentProcessId();
+
+    hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
```

Comment:
Damian, I tried it. But it seems the only way to get parent process id is to use the two parameters . I tried with pid. Did not work because the getfirstprocess function does not return the current process  (as expected) - very strange.